js/matchingModel.js

```diff
@@ -0,0 +1,227 @@
+define([
+    'core/js/models/questionModel'
+], function(QuestionModel) {
+    
+    var MatchingModel = QuestionModel.extend({
+
+        init: function() {
+            QuestionModel.prototype.init.call(this);
+
+            this.setupQuestionItemIndexes();
+        },
+
+        setupQuestionItemIndexes: function() {
+
+            this.get('_items').forEach(function(item, index) {
+                if (item._index === undefined) {
+                    item._index = index;
+                    item._selected = false;
+                }
+                item._options.forEach(function(option, index) {
+                    if (option._index === undefined) {
+                        option._index = index;
+                        option._isSelected = false;
+                    }
+                });
+            });
+        },
+
+        setupRandomisation: function() {
+            if (!this.get('_isRandom') || !this.get('_isEnabled')) return;
+
+            this.get('_items').forEach(function(item) {
+                item._options = _.shuffle(item._options);
+            });
+        },
+
+        restoreUserAnswers: function() {
+            if (!this.get('_isSubmitted')) return;
+
+            var userAnswer = this.get('_userAnswer');
+
+            this.get('_items').forEach(function(item, index) {
+                item._options.forEach(function(option, index) {
+                    if (option._index == userAnswer[item._index]) {
```

Comment:
this double equals > triple equals is the only thing that jumped out.
could also be a return early.

:+1: